test/test_helper.ts

```diff
@@ -6,9 +6,13 @@ export const testHelper = {
    * @param promise The original promise.
    * @returns The inverted promise, which fulfills if rejected and rejects if fulfilled.
    */
-  async willBeRejected<ErrorT = any>(promise: Promise<any>): Promise<ErrorT> {
+  async willBeRejected<ErrorT = any>(promise: Promise<any> | (() => Promise<any>)): Promise<ErrorT> {
     try {
-      await promise;
+      if (typeof promise === 'function') {
```

Comment:
Nit, how come we need this? Should be able to just call the function at the call site if it truly returns a promise.

Reply:
Oops, I should have used `assert.throws()`. Removed this